lib/engine/matchers/all_keys_matcher.rb

```diff
@@ -15,7 +15,7 @@ def initialize
     # @param key [string] key value to be matched
     #
     # @return [boolean] true for all instances
-    def match?(_key, _attributes)
+    def match?(_key, _split_treatment, _attributes)
```

Comment:
Small comment, but I think the correct signature for the match function would be (key, evaluator, attributes).